sequential/seq2pat.py

```diff
@@ -151,6 +156,12 @@ def values(self):
         """
         return self._values
 
+    def _set_values(self, values):
```

Comment:
This function was not needed before, but now necessarily. Can you add a comment inside to say why is this needed and where it is used (sth like to re-set the constraints within batch etc.. )

Reply:
Yes, this function is needed when we split the data into batches. 

In each batch, we need to set the correct attributes and the constraints. Since the upper and lower bound are set on the entire set, it would be easier to copy these constraints but only reset the attribute values in a batch. 

I am prevented to set the values directly, so I add the set_value function. I will add a comment inside.

Reply:
exactly! Commenting the above in the code would be great